lib/chef/knife/mixin/helper.rb

```diff
@@ -43,6 +44,35 @@ def values_from_json(json)
       rescue JSON::ParserError
         raise JSON::ParserError, "#{json} is not valid JSON!"
       end
+
+      # I/P: json string
+      # It checks wheather it contains any non-printable character present or not,
+      # If present then raises error of InvalidValue
+      def validate_json(json)
+        begin
+          evaled_json = eval(json)
+        rescue StandardError => e
+          puts e.message
+        end
+
+        if evaled_json.is_a?(Hash)
+          evaled_json.each do |key, value|
+            next if printable?(value)
+
+            msg = "Value '#{value}' of key '#{key}' contains non-printable characters. Check that backslashes are escaped with another backslash (e.g. C:\\\\Windows) in double-quoted strings."
+            raise ChefVault::Exceptions::InvalidValue, msg
+          end
+        end
```

Comment:
This will fail unless all of the values of the json are strings. That is, `{ ids: [1,2,3] }`, or `{ id: 2 }` will cause this to fail. Will the json always be in that format, or should this code try to handle those cases?

Reply:
yes @phiggins, it will work only for string values. any way to handle other formats?